src/Workspaces/Core/Portable/Execution/CustomAsset.cs

```diff
@@ -62,12 +62,21 @@ private static Checksum CreateChecksumFromStreamWriter(WellKnownSynchronizationK
     /// </summary>
     internal sealed class WorkspaceAnalyzerReferenceAsset : CustomAsset
     {
+        // host analyzer is not shadow copied, no need to load assembly to get real path
+        // this also prevent us from loading assemblies for all vsix analyzers preemptively
+        private const bool usePathFromAssembly = false;
+
         private readonly AnalyzerReference _reference;
         private readonly ISerializerService _serializer;
 
-        public WorkspaceAnalyzerReferenceAsset(AnalyzerReference reference, ISerializerService serializer) :
+        public WorkspaceAnalyzerReferenceAsset(
+            AnalyzerReference reference,
+            ISerializerService serializer,
+            IReferenceSerializationService hostSerializationService) :
             base(
-                serializer.CreateChecksum(reference, CancellationToken.None),
+                Checksum.Create(
+                    WellKnownSynchronizationKind.AnalyzerReference,
+                    hostSerializationService.CreateChecksum(reference, usePathFromAssembly, CancellationToken.None)),
```

Comment:
📝 If cancellation is desired on this path, it should be a parameter rather than expect the caller to check the cancellation token and then ignore it. This is the location I was referencing from another comment.

Reply:
sure.